App/Sources/Feature/StudentSignUpFeature/StudentSignUpViewModel.swift

```diff
@@ -0,0 +1,196 @@
+import Foundation
+import Service
+
+final class StudentSignUpViewModel: ObservableObject {
+    @Published var schoolSearch = ""
+    @Published var clubSearch = ""
+    @Published var name = ""
+    @Published var yearOfAdmission = ""
+    @Published var studentID = "" {
+        didSet {
+            parseStudentID()
+        }
+    }
+    @Published var phoneNumber = "" {
+        didSet {
+            parsePhoneNumber()
+        }
+    }
+    @Published var certificationNumberPhoneNumber = ""
+    @Published var email = ""
+    @Published var certificationNumberEmail = ""
+    @Published var password = ""
+    @Published var checkPassword = ""
+    @Published var phoneNumberTimeRemaining : Int = 180
+    @Published var emailTimeRemaining : Int = 180
+    @Published var selectedSchool: String = "학교"
+    @Published var selectedClub: String = "동아리"
+    @Published var clubsForSelectedHighSchool: [String] = []
+    let highSchool: [HighSchoolType] = HighSchoolType.allCases
+    private var timer: Timer?
+    
+    var getClubsForSelectedHighSchool: HighSchoolType? {
+        didSet {
+            clubsForSelectedHighSchool = getClubsForSelectedHighSchool?.getClubsForSelectedHighSchool() ?? []
+                }
+        }
+    
+    public func parseStudentID() {
+        guard studentID.count == 4, let grade = Int(String(studentID.prefix(1))), let classNumber = Int(String(studentID.dropFirst(1).prefix(1))), let studentNumber = Int(String(studentID.suffix(2))) else { return }
```

Comment:
```suggestion
        guard studentID.count == 4, 
                    let grade = Int(String(studentID.prefix(1))),
                    let classNumber = Int(String(studentID.dropFirst(1).prefix(1))),
                    let studentNumber = Int(String(studentID.suffix(2))) 
        else { return }
```
한 줄이 120자 이상 넘어가면 이런식으로 개행해주세요